src/effects/onReplace/index.js

```diff
@@ -0,0 +1,23 @@
+import validate from '../validate';
+
+function onReplace() {
+  return validate({
+    name: 'onReplace',
+    realTarget: action => action.target,
+    do: (action, state) => {
+      const { payload, index, condition, target } = action;
+      let array = [...state[target]];
+      if (index) array[index] = payload;
+      else if (condition) {
+        const arrayFilter = array.filter(condition);
```

Comment:
I'd use `.reduce` for this. For each item, you can choose to replace it with `payload` or leave it as is